extensions/sld/include/powsybl/iidm/extensions/sld/ConnectablePositionAdder.hpp

```diff
@@ -0,0 +1,116 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#ifndef POWSYBL_IIDM_EXTENSIONS_SLD_CONNECTABLEPOSITIONADDER_HPP
+#define POWSYBL_IIDM_EXTENSIONS_SLD_CONNECTABLEPOSITIONADDER_HPP
+
+#include <powsybl/iidm/ExtensionAdder.hpp>
+#include <powsybl/iidm/extensions/sld/ConnectablePosition.hpp>
+#include <powsybl/stdcxx/optional.hpp>
+
+namespace powsybl {
+
+namespace iidm {
+
+namespace extensions {
+
+namespace sld {
+
+class ConnectablePositionAdder : public ExtensionAdder {
+public:
+    class FeederAdder {
+    public:
+        FeederAdder(ConnectablePositionAdder& parent, ConnectablePosition::OptionalFeeder& feeder);
+
+        ConnectablePositionAdder& add();
+
+        FeederAdder& withDirection(const ConnectablePosition::Direction& direction);
+
+        FeederAdder& withName(const std::string& name);
+
+        FeederAdder& withOrder(unsigned long order);
+
+    private:
+        std::string m_name;
+
+        stdcxx::optional<unsigned long> m_order;
+
+        stdcxx::optional<ConnectablePosition::Direction> m_direction;
+
+        std::reference_wrapper<ConnectablePositionAdder> m_parent;
+
+        std::reference_wrapper<ConnectablePosition::OptionalFeeder> m_feeder;
```

Comment:
This is strange and not compatible with the expected design?

Reply:
Non compatible with the expected design => Agree
Strange : to be discussed :). In Java implementation, it uses a template specialization on-the-fly. Not sure it could be done that simple in C++.